synapse/media/url_previewer.py

```diff
@@ -329,7 +301,8 @@ async def _do_preview(self, url: str, user: UserID, ts: int) -> bytes:
                 # defer to that.
                 oembed_url = self._oembed.autodiscover_from_html(tree)
                 og_from_oembed: JsonDict = {}
-                if oembed_url:
+                # Only download to the oEmbed URL if it is allowed.
+                if oembed_url and not self._is_url_blocked(oembed_url):
                     try:
                         oembed_info = await self._handle_url(
```

Comment:
It looks like we could move all of the `self._is_url_blocked(...)` checks into `self._handle_url(...)` since we always run the check before passing it there anyway.

Reply:
I looked at this, but I don't think it works because the behavior for a blocked URL is different. (If the original URL is blocked we raise an exception vs. if a follow-up URL is blocked we skip that enrichment.)

Reply:
Actually, it looks like we might be able to just raise an error since for oEmbed / pre-caching images we catch exceptions anyway.